website/public/js/controllers/groupsCtrl.js

```diff
@@ -224,6 +224,9 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       if ($scope.query === undefined || $scope.query === null) {
         return false;
       }
+      if (msg.username == undefined) {
+        return false;
```

Comment:
Ahh, you're right, it's `$scope.query.text` that is important. 

You can check the type of data a variable is with `typeof`, for example: `typeof msg.username === 'string'`. I have mixed feelings about using it though.

Can we trust that if `$scope.query` exists, that `$scope.query.text` exists and is a string? And if `msg.user` exists, it is a string?